packages/core/src/tracing/instrumentation/protocols/httpClient.js

```diff
@@ -19,17 +19,24 @@ const {
 const constants = require('../../constants');
 const cls = require('../../cls');
 const url = require('url');
+const requireHook = require('../../../util/requireHook');
+let logger;
+logger = require('../../../logger').getLogger('tracing/httpClient', newLogger => {
+  logger = newLogger;
+});
 
 let extraHttpHeadersToCapture;
 let isActive = false;
 
 exports.init = function init(config) {
   instrument(coreHttpModule, false);
   instrument(coreHttpsModule, true);
-
   extraHttpHeadersToCapture = config.tracing.http.extraHttpHeadersToCapture;
+  requireHook.onModuleLoad('request', logDeprecatedWarning);
```

Comment:
Suggestion: Maybe we should ensure that we only print the warning once? The `request` module is used in lot's of sub dependencies.

Reply:
In app.js:

> require('request') -> uncached
> require('request') -> cached

This should trigger the warning twice.

Reply:
Verified and it logged once.